### PR TITLE
Switch Annotation to attributes

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Switch listener to attributes for `@DemoRestricted`
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI 🟢 + automated tests

### CODE REVIEW

File: src/PrestaShopBundle/Security/Annotation/DemoRestricted.php [hunk 1]
```diff
@@ -63,6 +60,18 @@ class DemoRestricted extends ConfigurationAnnotation
...
+        ?string $redirectRoute = null,
+        string $domain = 'Admin.Notifications.Error',
+        string $message = 'This functionality has been disabled.',
+        array $redirectQueryParamsToKeep = []
+    ) {
+        $this->domain = $domain;
+        $this->message = $message;
+        $this->redirectRoute = $redirectRoute;
+        $this->redirectQueryParamsToKeep = $redirectQueryParamsToKeep;
+    }
```
Comment (boherm) [inline src/PrestaShopBundle/Security/Annotation/DemoRestricted.php]: Do you think that we can use constructor property promotion on this?

**Review by boherm — DISMISSED** [review]
LGTM, thanks @FabienPapet to make this good improvement!
Just one comment but I don't block this PR for this.

File: src/PrestaShopBundle/Security/Annotation/DemoRestricted.php (hunk 1 above)
Comment (FabienPapet) [reply src/PrestaShopBundle/Security/Annotation/DemoRestricted.php]: Sure, I updated the code

**Review by boherm — APPROVED** [review]
LGTM, thanks @FabienPapet!

**Review by tleon — APPROVED** [review]

**Review by djoelleuch — APPROVED** [review]
Hello @FabienPapet 
automated tests : OK (https://github.com/djoelleuch/ga.tests.ui.pr/actions/runs/5455935371/jobs/9928969106)
It's QA approved :heavy_check_mark:
Thank you :rocket:

**Comment by mflasquin** [comment]
Thanks @FabienPapet

**Pull request merged**